tools/autograd/gen_python_functions.py

```diff
@@ -66,6 +68,7 @@
 PY_VARIABLE_METHOD_NOARGS = CodeTemplate("""\
 static PyObject * ${pycname}(PyObject* self_, PyObject* args)
 {
+  ${deprecation_message}
```

Comment:
hmm this is interesting, because removing these won't actually let us get rid of the methods, because this is only affecting the python API (and it currently must at least include this because the python functions can call C++ methods, so you can't put the deprecation warning in the C++ methods).